### ISSUE TITLE
Modifier "Variation" not working right

### ISSUE THREAD

**Issue by pavoltanuska** [firesale/firesale#394]
Version: 1.2/develop

I'm unable to insert variations in modifier "Variation".

Steps to reproduce:
- product admin - tab Modifiers
- Modifier "Variation" added successfuly
- click on "Add a Variation"
- overlay pops out
- write title, price modifier, click "Save"

Result:
- error "Internal Server Error" appears
- response from controller handling ajax request is:

---

```
A Database Error Occurred
Error Number: 1054

Unknown column 'default_fp_0.firesale_products_id' in 'field list'

SELECT `default_fp_0`.`firesale_products_id`FROM `default_firesale_product_variations_firesale_products` AS `fp_0`WHERE `fp_0`.`row_id` = '25'AND `fp_0`.`firesale_product_variations_id` = '21'
```

---

```
A Database Error Occurred
Error Number: 1054

Unknown column 'fp_0.row_id' in 'where clause'

UPDATE `default_ci_sessions` SET `last_activity` = 1373478080, `user_data` = 'a:9:{s:8:\"username\";s:7:\"username\";s:5:\"email\";s:14:\"email@email.email\";s:2:\"id\";s:1:\"2\";s:7:\"user_id\";s:1:\"2\";s:8:\"group_id\";s:1:\"1\";s:5:\"group\";s:5:\"admin\";s:8:\"category\";s:0:\"\";s:13:\"cart_contents\";a:3:{s:32:\"fc58d335c531193b11f89e6d0d4312c5\";a:14:{s:2:\"id\";s:2:\"17\";s:4:\"code\";s:32:\"product-1-A2\";s:3:\"qty\";d:1;s:5:\"price\";d:120;s:8:\"tax_band\";s:1:\"1\";s:4:\"name\";s:40:\"Product 1 - a2\";s:4:\"slug\";s:38:\"product-1-a2\";s:4:\"ship\";s:1:\"1\";s:6:\"weight\";s:0:\"\";s:5:\"image\";b:0;s:7:\"options\";a:1:{i:4;a:6:{s:6:\"mod_id\";s:1:\"4\";s:6:\"var_id\";s:2:\"21\";s:4:\"type\";s:1:\"1\";s:5:\"title\";s:4:\"varz\";s:5:\"value\";s:2:\"a2\";s:5:\"price\";s:1:\"0\";}}s:6:\"parent\";s:1:\"2\";s:5:\"rowid\";s:32:\"fc58d335c531193b11f89e6d0d4312c5\";s:8:\"subtotal\";d:120;}s:10:\"cart_total\";d:120;s:11:\"total_items\";d:1;}s:17:\"flash:new:success\";s:34:\"New variation created successfully\";}' WHERE `fp_0`.`row_id` = '25' AND `fp_0`.`firesale_product_variations_id` = '21' AND `session_id` = '1075fb809cacba81b3170758862572fb'
```

---

Possible cause:
- column names in select (but aliases too) are prepended with SITE_REF (in this case "default") in PyroCMS.

Suggested fix:
- in firesale/models/modifier_m.php refactor method variation_exists(). Prepend all "fp_0" aliases with SITE_REF constant.
- Lines: 461, 467, 472, 473

Edit: formatting

**Comment by jHoldroyd** [comment]
This should have been fixed by 766747e - if you could update and try again that would be great. And sorry for the issue.

**Comment by pavoltanuska** [comment]
updated, now it works as expected. thanks!
